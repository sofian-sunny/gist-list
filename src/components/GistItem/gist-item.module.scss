.container {
    flex: 1;
    width: 70%;
    margin-top: 40px;
    margin-left: 15%;
    margin-right: 15%;
  }
  .rowItem {
    display: flex;
    flex-wrap: wrap;
    flex-direction: row;
    background-color: #f1f2f4;
    padding: 10px;
  }