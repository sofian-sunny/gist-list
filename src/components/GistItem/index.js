import React from "react";
import PropTypes from 'prop-types'
import cn from './gist-item.module.scss'

/**
 * This is SearchForm Method
 * this shows search bar in the Home page
 * @reactProps {string} title - welcome message
 */
const GistItem = (props) => {
    const { item } = props
    const { id, created_at, description, comments, owner } = item
    const { avatar_url, login } = owner

    const renderText = (firstValue, secondValue) => {
        return (
            <div className={cn.rowItem}>
                <div style={{ flex: 1 }}><p>{firstValue}</p></div>
                <div style={{ flex: 1 }}><p>{secondValue}</p></div>
            </div>
        )
    }

    return (
        <div className={cn.container}>
            <div className={cn.rowItem}>
                <img style={{ width: 50, height: 50, borderRadius: 25 }} src={avatar_url} alt="new" />
                <div style={{ flex: 1 }}>
                    {renderText(`Id: ${id}`, `Name: ${login}`)}
                    {renderText(`Created at: ${created_at}`, `Comments: ${comments}`)}
                    {renderText(`Description: ${description}`)}
                </div>
            </div>
        </div>
    )
}

GistItem.propTypes = {
    item: PropTypes.shape({
        id: PropTypes.string,
        created_at: PropTypes.string,
        description: PropTypes.string,
        comments: PropTypes.string,
        owner: PropTypes.shape({
            login: PropTypes.string,
            avatar_url: PropTypes.string,
        })
    })
}

GistItem.defaultProps = {
    item: {
        id: '#',
        created_at: '',
        description: '',
        comments: '',
        owner: {
            login: '',
            avatar_url: '',
        }
    }
}

export default GistItem
