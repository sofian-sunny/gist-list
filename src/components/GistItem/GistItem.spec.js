import React from 'react'
import { shallow } from 'enzyme'
import GistItem from './index'

const setUp = (props) => {
  const component = shallow(<GistItem {...props} />)

  return component
}

describe('GistItem Component render without error', () => {
  const props = {
    title: 'greetings title'
  }
  const wrapper = setUp(props)

  it('GistItem method should return list item with image', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
