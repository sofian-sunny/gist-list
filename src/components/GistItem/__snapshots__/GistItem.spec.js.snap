// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GistItem Component render without error GistItem method should return list item with image 1`] = `ShallowWrapper {}`;
